modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -81,23 +81,23 @@ export interface DateBucket
     to?: string;
 }
 
-export interface BucketsAggregation {
+export interface BucketsAggregationResult {
     buckets: (DateBucket | NumericBucket)[];
 }
```

Comment:
I mentioned earlier that it is a bit difficult to split `AggregationsResult`, since it's not a discriminate union (yet). 

And here we have another union (`DateBucket | NumericBucket`) inside it again, which is also a bit difficult to split. 

I would suggest changing this code, so that we only need to do one splitting of unions, and that is on the `AggregationsResult` level.

```suggestion
export interface DateBucketsAggregationResult {
    buckets: DateBucket[];
}

export interface NumberBucketsAggregationResult {
    buckets: NumericBucket[];
}
```

and then later:

```typescript
export type AggregationsResult = 
  | DateBucketsAggregationResult 
  | NumberBucketsAggregationResult 
  | StatsAggregationResult 
  | SingleValueMetricAggregationResult;
```

Reply:
Not sure we can split those buckets. 
@rymsha can we return the mixed buckets, list of both `DateRangeBucket` and `NumericRangeBucket`? It's hard to say from the [mapper](https://github.com/enonic/xp/blob/master/modules/lib/lib-node/src/main/java/com/enonic/xp/lib/node/mapper/AggregationMapper.java#L70)? We can return, at least.